app/Jobs/ElasticSearchIndexDelete.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace App\Jobs;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use App\WikiSetting;
+use App\Http\Curl\CurlRequest;
+use App\Http\Curl\HttpRequest;
+
+class ElasticSearchIndexDelete extends Job implements ShouldBeUnique
+{
+    private $wikiDomain;
+    private $wikiId;
+
+    private $request;
+
+    /**
+     * @return void
+     */
+    public function __construct( string $wikiDomain, int $wikiId, HttpRequest $request = null )
+    {
+        $this->wikiDomain = $wikiDomain;
+        $this->wikiId = $wikiId;
+        $this->request = $request ?? new CurlRequest();
+    }
+
+    /**
+     * The unique ID of the job.
+     *
+     * @return string
+     */
+    public function uniqueId()
+    {
+        return $this->wikiDomain;
+    }
+
+    /**
+     * @return void
+     */
+    public function handle()
+    {
+        $setting = WikiSetting::where([ 'wiki_id' => $this->wikiId, 'name' => WikiSetting::wwExtEnableElasticSearch, ])->first();
+
+        // job got triggered but no setting in database
+        if ( $setting === null ) {
+            $this->fail(
+                new \RuntimeException('wbstackElasticSearchDelete call for '.$this->wikiDomain.' was triggered but not setting available.')
+            );
+
+            return;
+        }
+        $elasticSearchBaseName = $this->wikiDomain;
+        $elasticSearchHost = getenv('ELASTICSEARCH_HOST');
+        
+        if( !$elasticSearchHost ) {
+            $this->fail( new \RuntimeException('ELASTICSEARCH_HOST not configured') );
+            return;
+        }
+
+        $url = $elasticSearchHost."/_cat/indices/{$elasticSearchBaseName}*?v&s=index&h=index";
```

Comment:
So this is probably fine for now, but introduces another connection between the API and an a service that is only really intended for mediawiki at this point (elasticsearch).

In terms of https://wbstack.github.io/docs/assets/img/2021-tech-overview-gce-k8s.drawio.50fd9b06.svg this would introduce another arrow / cross dependency.
The only other major one of these right now is the creation of wiki dbs in the first place, and at some point we would want to push all of that to the MediaWiki side of things, and the API would just talk to mediawiki.